tests/components/tuya/test_climate.py

```diff
@@ -55,3 +59,81 @@ async def test_platform_setup_no_discovery(
     assert not er.async_entries_for_config_entry(
         entity_registry, mock_config_entry.entry_id
     )
+
+
+@pytest.mark.parametrize(
+    "mock_device_code",
+    ["kt_serenelife_slpac905wuk_air_conditioner"],
+)
+async def test_fan_mode_windspeed(
+    hass: HomeAssistant,
+    mock_manager: ManagerCompat,
+    mock_config_entry: MockConfigEntry,
+    mock_device: CustomerDevice,
+) -> None:
+    """Test fan mode with windspeed."""
+    await initialize_entry(hass, mock_manager, mock_config_entry, mock_device)
+
+    state = hass.states.get("climate.air_conditioner")
+    assert state is not None, "climate.air_conditioner does not exist"
+    assert state.attributes["fan_mode"] == WINDSPEED_LOW
+    await hass.services.async_call(
+        Platform.CLIMATE,
+        "set_fan_mode",
+        {
+            "entity_id": "climate.air_conditioner",
+            "fan_mode": WINDSPEED_HIGH,
+        },
+    )
+    await hass.async_block_till_done()
+    mock_manager.send_commands.assert_called_once_with(
+        mock_device.id, [{"code": "windspeed", "value": "2"}]
+    )
+
+    # Simulate the device reporting the new windspeed
+    mock_device.status["windspeed"] = WINDSPEED_HIGH
+    await hass.services.async_call(
+        "homeassistant",
+        "update_entity",
+        {"entity_id": "climate.air_conditioner"},
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.air_conditioner")
+    assert state is not None, (
+        "climate.air_conditioner does not exist after service call"
+    )
+    assert state.attributes["fan_mode"] == WINDSPEED_HIGH
```

Comment:
I don't think this part "Simulate the device reporting the new windspeed" is needed in the same test

I think it should be a separate test, using the device listener similar to https://github.com/home-assistant/core/pull/125098/files#diff-ba8c8bfcb09fab51857cdd9f286698a867da7a456abeee7cdfb04734b830f9f1

And it can be done in a follow-up PR

Reply:
```suggestion
```